chopper.py

```diff
@@ -229,189 +267,192 @@ def split_by_columns(filepath: Path, col_list: list[str], config: Config) -> lis
     for file in files.values():
         file["fout"].close()
 
-    if not config.is_original:
-        os.remove(filepath)
+    if filepath.absolute().parents[0] == config.output_directory.absolute():
+        filepath.unlink()
 
     return list(files.keys())
 
 
-def split_by_equal(filepath: Path, equal: int, config: Config) -> list[Path]:
+def split_by_equal(filepath: Path, equal: int, config: ChopperNamespace) -> list[Path]:
     """Creates N files of approximately (+/- 1) equal size.
 
     Args:
         filepath (Path): The path of the file to split.
         equal (int): Number of files to split into.
-        config (Config): Config object created from CLI arguments.
+        config (ChopperNamespace): Namespace object created from CLI arguments.
 
     Returns:
         list[Path]: List of split intermediate files.
     """
 
-    with open(filepath, "r", encoding=config.encoding) as fin:
-        files = [config.output_dir / f"{filepath.stem}_{i+1}" for i in range(equal)]
-        fouts = [open(f, "w", encoding=config.encoding) for f in files]
-        fout_cycle = cycle(fouts)
+    with filepath.open("rb", encoding=config.encoding) as fin_obj:
+        with mmap.mmap(fin_obj.fileno(), length=0, **config.mmap_kwargs) as fin:
+            if os.name != "nt":
+                fin.madvise(mmap.MADV_SEQUENTIAL)
 
-        for i, row in enumerate(fin):
-            if i == 0:
-                for fout in fouts:
-                    fout.write(row)  # Write headers
-                continue
+            paths = [
+                config.output_directory / f"{filepath.stem}_{i+1}" for i in range(equal)
+            ]
+            fouts = [f.open("wb", encoding=config.encoding) for f in paths]
+            fout_cycle = cycle(fouts)
 
-            next(fout_cycle).write(row)
+            headers = fin.readline()
+            for fout in fouts:
+                fout.write(headers)
+            while row := fin.readline():
+                next(fout_cycle).write(row)
```

Comment:
I changed a few of the for loops with counters and such over to while loops, which look a bit more elegant.